options.py

```diff
@@ -45,6 +45,14 @@ class BaseOption(ABC):
     def __init__(self) -> None:
         raise NotImplementedError
 
+    @abstractmethod
+    def _to_json(self) -> JSON:
+        raise NotImplementedError
+
+    @abstractmethod
+    def _from_json(self, value: JSON) -> None:
+        raise NotImplementedError
```

Comment:
this is still somewhat of an interface, so good to add docstrings - especially to explain ellipsis
```
Turns this option into a JSON value.

Returns:
    This option's JSON representation, or Ellipsis if it should be considered to have no value.
```
```
Assigns this option's value, based on a previously retrieved JSON value.

Args:
    value: The JSON value to assign.
```